### ISSUE TITLE
`fundchannel_complete` crashes on peer_error?

### ISSUE THREAD

**Issue by niftynei** [ElementsProject/lightning#2831]
### Issue and Steps to Reproduce

Crash when calling `fundchannel_complete`.

```
+59249.972233100 lightningd(25401):DEBUG: lightning_openingd-02...4a chan #40: peer_in WIRE_ERROR
+59249.972659597 lightningd(25401):INFO: 02...4a chan #40: channel 5c...f8: funding failed due to internal error                           
+59250.101029186 lightningd(25401):BROKEN: FATAL SIGNAL 11 (version v0.7.1-64-gbb30104)
+59250.101053244 lightningd(25401):BROKEN: backtrace: common/daemon.c:45 (send_backtrace) 0x56450653b7ae
+59250.101067785 lightningd(25401):BROKEN: backtrace: common/daemon.c:53 (crashdump) 0x56450653b7fe
+59250.101077871 lightningd(25401):BROKEN: backtrace: (null):0 ((null)) 0x7fb50c7bef1f
+59250.101090137 lightningd(25401):BROKEN: backtrace: lightningd/jsonrpc.c:539 (json_stream_raw_for_cmd) 0x564506513b46 
+59250.101102902 lightningd(25401):BROKEN: backtrace: lightningd/jsonrpc.c:561 (json_start) 0x564506513c76
+59250.101115643 lightningd(25401):BROKEN: backtrace: lightningd/jsonrpc.c:580 (json_stream_fail_nodata) 0x564506513d2c 
+59250.101127982 lightningd(25401):BROKEN: backtrace: lightningd/jsonrpc.c:498 (command_fail) 0x5645065139c8
+59250.101137717 lightningd(25401):BROKEN: backtrace: lightningd/opening_control.c:103 (uncommitted_channel_disconnect) 0x56450651b110
+59250.101149916 lightningd(25401):BROKEN: backtrace: lightningd/opening_control.c:719 (opening_channel_errmsg) 0x56450651cda1
+59250.101160025 lightningd(25401):BROKEN: backtrace: lightningd/subd.c:383 (handle_peer_error) 0x564506534618
+59250.101171846 lightningd(25401):BROKEN: backtrace: lightningd/subd.c:464 (sd_msg_read) 0x564506534939
+59250.101183525 lightningd(25401):BROKEN: backtrace: lightningd/subd.c:303 (read_fds) 0x56450653434e
+59250.101197954 lightningd(25401):BROKEN: backtrace: ccan/ccan/io/io.c:59 (next_plan) 0x5645065809da
+59250.101207847 lightningd(25401):BROKEN: backtrace: ccan/ccan/io/io.c:407 (do_plan) 0x564506581557
+59250.101218748 lightningd(25401):BROKEN: backtrace: ccan/ccan/io/io.c:417 (io_ready) 0x564506581595
+59250.101229802 lightningd(25401):BROKEN: backtrace: ccan/ccan/io/poll.c:445 (io_loop) 0x5645065835ea
+59250.101238238 lightningd(25401):BROKEN: backtrace: lightningd/io_loop_with_timers.c:24 (io_loop_with_timers) 0x564506510d4f
+59250.101249982 lightningd(25401):BROKEN: backtrace: lightningd/lightningd.c:833 (main) 0x564506516c46
+59250.101261181 lightningd(25401):BROKEN: backtrace: (null):0 ((null)) 0x7fb50c7a1b96
+59250.101271388 lightningd(25401):BROKEN: backtrace: (null):0 ((null)) 0x5645064ffc29
+59250.101280487 lightningd(25401):BROKEN: backtrace: (null):0 ((null)) 0xffffffffffffffff
```

### `getinfo` output

Reported-By: @rsbondi

**Comment by rsbondi** [comment]
I have some additional information, I attempted this again, and I was having issues crashing shortly after `fundchannel_start` this time around(maybe last time, just didn't notice until `fundchannel_complete`).  

The crash seemed to occur when connecting to peers of unknown implementation, once I connected to a known clightning user, everything was fine.  The peers where I had issues were lnbig and satoshilabs, not sure there is a way to know what implementation they run.
